### PR TITLE
b/157610728: add consumer-type and consumer-number headers

### PR DESCRIPTION
**Pull request by TAOXUY**
- Add header of Prefix-Consumer-Type and header of Prefix-Consumer-Number.

- Header Prefix-Project-Number is removed since it's been replaced by Consumer-Number. Though it is a breaking change, we think no many users are using this so it should be fine.

### CODE REVIEW

File: src/api_proxy/service_control/request_builder.cc [hunk 1]
```diff
@@ -1364,6 +1366,19 @@ Status RequestBuilder::ConvertCheckResponse(
...
   }
 
+  if (check_response.check_info().consumer_info().consumer_number() > 0) {
+    check_response_info->consumer_number = std::to_string(
+        check_response.check_info().consumer_info().consumer_number());
+  }
+
+  if (check_response.check_info().consumer_info().type() !=
+      CheckResponse_ConsumerInfo_ConsumerType::
+          CheckResponse_ConsumerInfo_ConsumerType_CONSUMER_TYPE_UNSPECIFIED) {
```
Comment (nareddyt) [inline src/api_proxy/service_control/request_builder.cc:1376]: Would prefer `CheckResponse::ConsumerInfo::ConsumerTypeCase::CONSUMER_TYPE_UNSPECIFIED`. Then you don't need the extra using statement either.

File: src/api_proxy/service_control/request_builder.cc [hunk 2]
```diff
@@ -1364,6 +1366,19 @@ Status RequestBuilder::ConvertCheckResponse(
...
+    check_response_info->consumer_type =
+        CheckResponse_ConsumerInfo_ConsumerType_Name(
```
Comment (qiwzhang) [inline src/api_proxy/service_control/request_builder.cc:1378]: Is this from proto generated code?   is this a exposed public functions or just internal implementation.  if latter,  proto team can remove it anytime, we will break.

File: src/api_proxy/service_control/request_info.h [hunk 3]
```diff
@@ -117,6 +117,10 @@ struct CheckResponseInfo {
   ScResponseErrorType error_type = ScResponseErrorType::ERROR_TYPE_UNSPECIFIED;
 
   std::string consumer_project_number;
```
Comment (nareddyt) [inline src/api_proxy/service_control/request_info.h:119]: Is this still needed?

Comment (TAOXUY) [reply src/api_proxy/service_control/request_info.h:119]: Yes, we have a label still using consumer_project_number
https://github.com/GoogleCloudPlatform/esp-v2/blob/585f37109ee2cf831c97f279f0a3755572b32bb8/src/api_proxy/service_control/request_builder.cc#L595

File: src/api_proxy/service_control/request_builder.cc (hunk 2 above)
Comment (TAOXUY) [reply src/api_proxy/service_control/request_builder.cc:1378]: It is generated. 

https://github.com/googleapis/googleapis/blob/8c53b2cb792234354c13336ac7daee61333deade/google/api/servicecontrol/v1/service_controller.proto#L122

Comment (TAOXUY) [reply src/api_proxy/service_control/request_builder.cc:1378]: Follow the convention of nestedProtoDefinition + _name

File: src/api_proxy/service_control/request_builder.cc (hunk 1 above)
Comment (TAOXUY) [reply src/api_proxy/service_control/request_builder.cc:1376]: That is so long.
Also consider both the enum and the util function converting enum to string need that namespace, so it is fine to add `using`.

**Pull request merged**